java/java.api.common/src/org/netbeans/modules/java/api/common/queries/ModuleInfoAccessibilityQueryImpl.java

```diff
@@ -232,48 +236,49 @@ private Collection<FileObject[]> collectModuleRoots(
         final MultiModule model = MultiModule.getOrCreate(mods, src);
         for (String modName : model.getModuleNames()) {
             final ClassPath cp = model.getModuleSources(modName);
-            res.add(cp.getRoots());
+            if (cp != null) {
+                res.add(cp.getRoots());
+            }
         }
         return res;
     }
 
     @NonNull
-    private static Optional<Pair<Set<FileObject>,Set<FileObject>>> readExports(
-            @NonNull final FileObject[] roots,
-            @NonNull final Set<? super FileObject> rootsCollector) {
-        Collections.addAll(rootsCollector, roots);
-        final Optional<FileObject> moduleInfo = Arrays.stream(roots)
-                .map((root) -> root.getFileObject(MODULE_INFO_JAVA))
-                .filter((mi) -> mi != null)
+    private static Optional<Pair<Set<FileObject>, Set<FileObject>>> extractExports(
+            @NonNull final FileObject[] roots) {
+        return Arrays.stream(roots)
+                .map(root -> root.getFileObject(MODULE_INFO_JAVA))
+                .filter(Objects::nonNull)
+                .map(mi -> {
+                    Set<FileObject> rootsSet = new HashSet<>();
+                    Collections.addAll(rootsSet, roots);
+                    Set<FileObject> exportsSet = readExports(mi, rootsSet);
+                    return Pair.of(rootsSet, exportsSet);
+                })
                 .findFirst();
```

Comment:
this is fine but a bit dangerous in case someone changes something in future (e.g adds sort or something like that).

moving the `map()` stage after `findFirst()` would be safer.

nitpick:
copy constructor candidate:
```java
                    Set<FileObject> rootsSet = new HashSet<>(roots);
```

Reply:
There isn't a constructor for HashSet` that takes an array (I don't think?) so I used `Set.of` instead - AFAICS the set is only ever read from so this is OK.